docs: add example on how to disable default routes for ssg (#30729)
nuxt/nuxt: docs/2.guide/1.concepts/3.rendering.md

The `200.html` and `404.html` might be useful for the hosting provider you are using.

> #### Skip default routes generation

Reviewer: ```suggestion
#### Skipping Client Fallback Generation
```

> When prerendering a client-rendered app, Nuxt will generate `index.html`, `200.html` and `404.html` files by default. However, if you need to prevent any (or all) of these files from being generated in your build, you can use the `'prerender:generate'` hook from [Nitro](/docs/getting-started/prerendering#prerendergenerate-nitro-hook).

Reviewer: ```suggestion
When prerendering a client-rendered app, Nuxt will generate `index.html`, `200.html` and `404.html` files by default. However, if you need to prevent any (or all) of these files from being generated in your build, you can use the `'prerender:generate'` hook from [Nitro](/docs/getting-started/prerendering#prerendergenerate-nitro-hook).

```

```ts twoslash [nuxt.config.ts]
export default defineNuxtConfig({
  ssr: false,
  nitro: {
    hooks: {
      'prerender:generate'(route) {
        const routesToSkip = ['/index.html', '/200.html', '/404.html']

>         if (routesToSkip.includes(route.route) ) {

Reviewer: ```suggestion
        if (routesToSkip.includes(route.route)) {
```

        }
      }
    }
  }
})
```

> 

Reviewer: ```suggestion
```

The `200.html` and `404.html` might be useful for the hosting provider you are using.

#### Skip default routes generation

> The default files `index.html`, `200.html` and `404.html` are part of Nuxt and often helpful. However, if you need to prevent these files from being generated in your build, you can use the `"prerender:generate"` hook from [Nitro](https://nuxt.com/docs/getting-started/prerendering#prerendergenerate-nitro-hook).

Reviewer: ```suggestion

When prerendering a client-rendered app, Nuxt will generate `index.html`, `200.html` and `404.html` files by default. However, if you need to prevent any (or all) of these files from being generated in your build, you can use the `'prerender:generate'` hook from [Nitro](/docs/getting-started/prerendering#prerendergenerate-nitro-hook).
```

In order to achieve this we can change how the routes are prerendered. Just add this to your hooks in your nuxt.config.ts:

```ts twoslash [nuxt.config.ts]
export default defineNuxtConfig({

>   ssr: false,
>   nitro: {
>     hooks: {
>       "prerender:generate"(route) {
>         const routesToSkip = ['/index.html', '/200.html', '/404.html']; 
>         if (routesToSkip.includes(route.route) ) {
>           route.skip = true;
>         }
>       }
>     },
>   }
> });

Reviewer: ```suggestion
  ssr: false,
  nitro: {
    hooks: {
      'prerender:generate'(route) {
        const routesToSkip = ['/index.html', '/200.html', '/404.html']
        if (routesToSkip.includes(route.route) ) {
          route.skip = true
        }
      }
    }
  }
})
```
